src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -0,0 +1,201 @@
+      this.timezoneOffset = parseInteger(timezoneOffsetMinutes) * 60;
+      int startTimeSeconds = getNumberOfSeconds(parseTime(startTime)) + this.timezoneOffset;
+      int endTimeSeconds = getNumberOfSeconds(parseTime(endTime)) + this.timezoneOffset;
+      if (startTimeSeconds > endTimeSeconds) {
+        throw new BadRequestException("Please enter a valid time!");
+      }
+      this.deliverySlot = new DeliverySlot(validateDate(deliveryDay, startTimeSeconds), startTimeSeconds, endTimeSeconds);
+      this.maxStops = parseInteger(maxStops);
+      if (this.maxStops > 25 || this.maxStops <= 0) {
+        throw new BadRequestException("Please enter between 1 and 25 stops!");
+      }
+    }
+    
+    /** 
+     * Sets the coordinates for the starting point of the delivery.
+     * TODO[ak47na]: represent latitude and longitude as a Point object.
+     */
+    public void setCoordinates(String latitude, String longitude) throws BadRequestException {
+      this.latitude = parseDouble(latitude);
+      this.longitude = parseDouble(longitude);
+    }
+    
+    /**
+     * Calls Geocoding API to find the latitude and longitude of the provided address, and sets 
+     * them as the starting point of the delivery.
+     */
+    public void setCoordinatesFromAddress(String startAddress) throws ApiException, InterruptedException, IOException, DataNotFoundException {
+      LatLng point = MapsRequest.getLocationFromAddress(startAddress);
+      if (point == null) {
+        // Geocoding API didn't find the coordinates corresponding to given address.
+        throw new DataNotFoundException("Unable to find address!");
+      }
+      this.latitude = point.lat;
+      this.longitude = point.lng;
+    }
+
+    /**
+     * Return a Date object from dateString.
+     */
+    private Date validateDate(String dateString, int startTimeSeconds) throws BadRequestException {
+      SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+      Date date = null;
+      try {
+        date = formatter.parse(dateString);
+      } catch (ParseException e) {
+        throw new BadRequestException("Please enter a valid date!");
+      }
+      // get the currentDate to check that date is not set for a past date
+      Date currentDay = new Date(System.currentTimeMillis());
+      if (date.compareTo(currentDay) < 0) {
+        throw new BadRequestException("Please enter a valid date!");
+      }
+      if (date.getTime() + startTimeSeconds * 1000 < currentDay.getTime()) {
+        throw new BadRequestException("Please enter a time in the future!");
+      }
+      return date;
+    }
+
+    /**
+     * Returns the integer value of numberString
+     */
+    private Integer parseInteger(String numberString) throws BadRequestException {
+      Integer number = null;
+      try {
+        number = Integer.parseInt(numberString);
+      } catch (Exception e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return number;
+    }
```

Comment:
What is the point of this?  Just a wrapper for parseInt?  This function throws an exception if there is a problem anyway, so I don't see much benefit from calling parseInteger vs parseInt since you need to handle the exception from either case right?

Or am I confused?

(same for parseDouble below)

Reply:
It was just a wrapper, but I removed it.